controllers/usersignup/usersignup_controller_test.go

```diff
@@ -2047,6 +2047,70 @@ func TestUserSignupWithExistingMURDifferentUserIDOK(t *testing.T) {
 	})
 }
 
+func TestUserSignupPropagatedClaimsSynchronizedToMURWhenModified(t *testing.T) {
+	// given
+	userSignup := commonsignup.NewUserSignup()
+	ready := NewGetMemberClusters(NewMemberClusterWithTenantRole(t, "member1", corev1.ConditionTrue))
+
+	r, req, _ := prepareReconcile(t, userSignup.Name, ready, userSignup,
+		commonconfig.NewToolchainConfigObjWithReset(t, testconfig.AutomaticApproval().Enabled(true)),
+		baseNSTemplateTier, deactivate30Tier)
+	InitializeCounters(t, NewToolchainStatus(
+		WithMetric(toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey, toolchainv1alpha1.Metric{
+			"1,external": 1,
+		}),
+		WithMetric(toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey, toolchainv1alpha1.Metric{
+			string(metrics.External): 1,
+		}),
+	))
+
+	// when - The first reconcile creates the MasterUserRecord
+	res, err := r.Reconcile(context.TODO(), req)
+
+	// then
+	require.NoError(t, err)
+	require.Equal(t, reconcile.Result{}, res)
+
+	// Lookup the user signup again
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, userSignup)
+	require.NoError(t, err)
+	assert.Equal(t, "approved", userSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey])
+
+	// We should now have a MUR
+	murtest.AssertThatMasterUserRecords(t, r.Client).HaveCount(1)
+
+	key := types.NamespacedName{
+		Namespace: test.HostOperatorNs,
+		Name:      userSignup.Name,
+	}
+	instance := &toolchainv1alpha1.UserSignup{}
+	err = r.Client.Get(context.TODO(), key, instance)
+	require.NoError(t, err)
+	assert.Equal(t, "approved", instance.Labels[toolchainv1alpha1.UserSignupStateLabelKey])
+
+	mur := toolchainv1alpha1.MasterUserRecord{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, &mur)
+	require.NoError(t, err)
```

Comment:
you could simplify this whole part, no need to go through reconciles, look at the `TestMigrateMur` test how it is done there, or just move it there